### PR TITLE
Convert nrf51, nrf52820, nrf52832, nrf52840, nrf5340, nrf9160 boards

### PR DESCRIPTION
**Pull request by nordicjm**
- [ ] nrf51 (Only 1 board is still outstanding, 96b_carbon_nrf51, this is because it is a dual SoC board and requires STM32 SoC v2 support to convert)
- [x] nrf52805
- [x] nrf52810
- [x] nrf52820
- [x] nrf52832
- [x] nrf52840
- [x] nrf5340
- [x] nrf9131
- [x] nrf9151
- [x] nrf9160
- [x] nrf9161

### CODE REVIEW

**Comment by tejlmand** [comment]
From compliance:
> Warning: boards/v2/laird_connect/bl5340_dvk/Kconfig.bl5340_dvk:7  please, no spaces at the start of a line
> ....

I think those Kconfigs would be nice to address before merging, to minimize number of commits / changes on each file during porting.

> get_maintainer.MaintainersError: MAINTAINERS.yml: glob pattern 'boards/arm/qemu_cortex_m0/' in 'files' in area 'ARM arch' does not match any files

Can be ignored, let's make a MAINTAINERS update in the branch when porting has completed.

File: boards/v2/ublox/ubx_bmd300eval/board.yml
Comment (tejlmand) [inline boards/v2/ublox/ubx_bmd300eval/board.yml:5]: minor, but seems a newline is missing (editor setting ?)

File: boards/v2/ublox/ubx_bmd300eval/Kconfig.ubx_bmd300eval
Comment (tejlmand) [inline boards/v2/ublox/ubx_bmd300eval/Kconfig.ubx_bmd300eval]: with soc series no longer a choice we should actually have the soc selecting the series and avoid the need for this selection. (something not possible in old hwm as you cannot select a choice entry in Kconfig)

File: boards/v2/ublox/ubx_bmd300eval/Kconfig.ubx_bmd300eval
Comment (tejlmand) [inline boards/v2/ublox/ubx_bmd300eval/Kconfig.ubx_bmd300eval]: not needed, this is done automatically with the benefit that if `config BOARD_UBX_BMD300EVAL` is mispelled we'll get a Kconfig warning.

Ref: https://github.com/zephyrproject-rtos/zephyr/blob/4a5473603d38877c0fdc54a7b1a5db203a0e0f51/boards/Kconfig.v2#L5-L16

https://github.com/tejlmand/zephyr/blob/3df5cfea48a7be424007de0fd9e1934a78360d76/boards/Kconfig.v2#L5-L16

File: boards/v2/nordic_nrf/nrf9160dk/board.yml
Comment (tejlmand) [inline boards/v2/nordic_nrf/nrf9160dk/board.yml:3]: what happened to revisions ?

File: boards/v2/raytac/raytac_mdbt53v_db_40/CMakeLists.txt
Comment (tejlmand) [inline boards/v2/raytac/raytac_mdbt53v_db_40/CMakeLists.txt:4]: this is board specific CMakeLists.txt, and only sourced when this board is building, so we could reduce check to:
```suggestion
if(CONFIG_SOC_NRF5340_CPUAPP_QKAA AND CONFIG_BOARD_ENABLE_CPUNET)
```

(not required for merging, but could be considered in case file has to be touched for other reasons)
Same comment goes for other cmake files in commit.

File: boards/v2/actinius/actinius_icarus/board.yml
Comment (tejlmand) [inline boards/v2/actinius/actinius_icarus/board.yml:3]: what about revisions ?

**Review by tejlmand — CHANGES_REQUESTED** [review]
Generally good looking 👍 .

Noticed a couple of minors which seems to be a repeating pattern, so only mention them once.

Only major that seems to need fixing before merging would be missing handling of board revisions.
The minor can be addressed later (but also done immediately if you like)

File: boards/v2/ublox/ubx_bmd300eval/board.yml
Comment (nordicjm) [reply boards/v2/ublox/ubx_bmd300eval/board.yml:5]: these were generated by the v1 -> v2 bash script :)

Comment (tejlmand) [reply boards/v2/ublox/ubx_bmd300eval/board.yml:5]: > (~editor~ script setting ?)

Fixed 😆

**Pull request merged**
